### DISCUSSION TITLE
Simulating weight distribution

### DISCUSSION THREAD

**Discussion by livPKPD** [LAPKB/Pmetrics#83] [General]
Hi there, I have a model of an oral drug that is dosed on a mg/kg basis. Weight as a covariate is not in the final model as it did not correlate with any PK parameters. 

I want to run some simulations - and for the simulated patients to have weights that reflect the distribution of weights in the study population. Do I do that in the simulation datafile?

Many thanks

**Comment by mnneely** [comment]
The easiest way to this is to put the dose in mg/kg in your simulation template. Assuming that WT is a covariate in your data, even if not used in the model, you can then add an #FA block (bioavailability) to your model as follows:

#FA
FA(1) = WT

This will multiply all oral doses by the WT. If you want to simulate weights, be sure to include the following code, assuming you are using the R6 Pmetrics and have loaded the model with `run1 <- PM_load(1)`, or some other number in the parentheses.

`covList = list(cov = run1$cov, mean = list(wt = *mu*), sd = list(wt = *sigma*), limits = list(wt = c(*lower, upper*))
sim1 <- PM_sim$run(run1$final, data = "template.csv", model = "new_model.txt", covariate = covList, ...)`

Of course, you replace the starred items with the appropriate values, and you can add additional simulation parameters to replace the "...", such as limits, predInt, nsim, etc.

**Reply by livPKPD** [reply]
Hi, many thanks for this. I have tried the following:
covList = list(
  cov = simrunwt$cov$data, 
  mean = list(wt = 50), 
  sd = list(wt = 8.24), 
  limits = list(wt = c(37, 83))
  )
sim1 <- PM_sim$run(simrunwt$final
                   , data = "simWT.csv"
                   , model = "genmodel.txt"
                   , covariate = covList
                   , limits = NA, nsim = 1000
                   )
                  
...where simrunwt <-PM_load(1)

However, I get this error:
_Error in match.names(clabs, names(xi)) : 
  names do not match previous names_

I have updated PMetrics

Any leads?!
Many thnaks

**Comment by livPKPD** [comment]
Hello again,

I am just following up on this - I feel I have tried *everything* but unfortunately I am getting the same error message:
_Error in match.names(clabs, names(xi)) :
names do not match previous names_

When I run the examples.R file provided in the recent PM course, the sim runs just fine, including with covariates. So I am not sure what to try next!

Many thanks in advance for any guidance

**Comment by Siel** [comment]
Fixed the prublem and pushed it to a new branch to further testing.
